### PR TITLE
fix(rich-text-editor): DLT-1752 remove custom parsing

### PR DESCRIPTION
**Pull request by braddialpad**
# fix(rich-text-editor): DLT-1752 remove custom parsing

## Obligatory GIF (super important!)

Obligatory GIF

## :hammer_and_wrench: Type Of Change

These types will increment the version number on release:

- [x] Fix

## :book: Jira Ticket

https://dialpad.atlassian.net/browse/DLT-1752

## :book: Description

- Removed the code we put in previously to parse mentions on programmatic input.
- Added an async wait into the mentions example.

## :bulb: Context

What we were doing here with the parsing is not really the way TipTap is supposed to work. TipTap fully supports doing this on it's own, however you have to pass in input that it understands which is HTML or JSON input. see: https://tiptap.dev/docs/editor/guide/output

To do this properly we have to change our backend to support TipTaps html or json output and use that to populate the input. I talked to Bianca and they have agreed to update the backend to be able to handle this. For now we will just not parse links / mentions live in the input which is actually the way it works currently in production.

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

For all Vue changes:

- [x] I have made my changes in Vue 2 and Vue 3. Note: you may sync your changes from Vue 2 to Vue 3 (or vice versa) using the `./scripts/dialtone-vue-sync.sh` script. Read docs here: [Dialtone Vue Sync Script](../packages/dialtone-vue3/.github/CONTRIBUTING.md#dialtone-vue-sync-script)

## :crystal_ball: Next Steps

Check if all TipTap formatting will work correctly after the server side changes have been made

## :link: Sources

https://tiptap.dev/docs/editor/guide/output
https://tiptap.dev/docs/editor/api/utilities/html#generate-html-from-json

### CODE REVIEW

File: packages/dialtone-vue2/components/rich_text_editor/mention_suggestion.js [hunk 1]
```diff
@@ -1,26 +1,53 @@
...
+    name: 'Bradley Hawkins',
+    avatarSrc: 'https://avatars.githubusercontent.com/u/13851061?s=460&u=1f1b5b0b5b2b2b2b2b2b2b2b2b2b2b2b2b2b2b&v=4',
+  },
+  {
+    id: 'julio.ortega',
+    name: 'Tico Ortega',
+    avatarSrc: 'https://avatars.githubusercontent.com/u/13851061?s=460&u=1f1b5b0b5b2b2b2b2b2b2b2b2b2b2b2b2b2b2b&v=4',
+  },
+  {
+    id: 'ignacio.Ropolo',
```
Comment (juliodialpad) [inline packages/dialtone-vue2/components/rich_text_editor/mention_suggestion.js]:
```suggestion
    id: 'ignacio.ropolo',
```

File: packages/dialtone-vue2/components/rich_text_editor/rich_text_editor.stories.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ import slashCommandSuggestion from './slash_command_suggestion';
 
 // Default Prop Values
 export const argsData = {
-  value: `I am not a standalone component, please use Message Input instead :v_tone3::robot:!`,
+  value: '<p>I am not a standalone component, please use Message Input instead<emoji-component code=":v_tone3:" id="270c-1f3fd"></emoji-component><emoji-component code=":robot:" id="1f916"></emoji-component> !</p>',
```
Comment (juliodialpad) [inline packages/dialtone-vue2/components/rich_text_editor/rich_text_editor.stories.js]: Are those `id` attributes necessary?

File: packages/dialtone-vue2/components/rich_text_editor/rich_text_editor.vue [hunk 3]
```diff
@@ -504,79 +502,9 @@ export default {
...
-
-      matches.forEach(match => {
-        const mention = suggestions.find(({ id }) => id === match[1]);
-        this.internalValue = this.internalValue.replace(`@${match[1]}`, ` <mention-component name="${mention.name}" id="${mention.id}"></mention-component>`);
-      });
-    },
-
-    destroyEditor () {
-      this.editor.destroy();
-    },
```
Comment (juliodialpad) [inline packages/dialtone-vue2/components/rich_text_editor/rich_text_editor.vue]: We don't need this function anymore? or we weren't using it anywere?

**Review by juliodialpad — COMMENTED** [review]
it's working pretty good! but I have some doubts regarding the changes: 

- Why are we removing the possibility to paste and input channels and mentions?.
- Should I do the same with emoji to keep the same behavior on all plugins?
- Should we update the example to output the HTML instead of plain text so we can copy and paste the output to parse the examples?

**Comment by juliodialpad** [comment]
Found kind of a bug but I'm not able to reproduce it consistently.

Sometimes you loose the keyboard navigation ability until you click an option from the dropdown, It happened sometimes when copy/pasting text and then trying to mention a channel/mention but not happening always so not sure if it's a bug or not.

File: packages/dialtone-vue2/components/rich_text_editor/mention_suggestion.js (hunk 1 above)
Comment (braddialpad) [reply packages/dialtone-vue2/components/rich_text_editor/mention_suggestion.js]: lol I swear I already fixed this... must have not committed it or something.

File: packages/dialtone-vue2/components/rich_text_editor/rich_text_editor.stories.js (hunk 2 above)
Comment (braddialpad) [reply packages/dialtone-vue2/components/rich_text_editor/rich_text_editor.stories.js]: hmm it's definitely an attribute in the emoji extension but doesn't seem like it's used.. I'll look into it a bit more.

**Comment by braddialpad** [comment]
> Sometimes you loose the keyboard navigation ability until you click an option from the dropdown, It happened sometimes when copy/pasting text and then trying to mention a channel/mention but not happening always so not sure if it's a bug or not.

Hmm I think I got this once or twice too.. Will try and repro it.

**Comment by braddialpad** [comment]
> * Why are we removing the possibility to paste and input channels and mentions?.

Because the backend doesn't understand the data that TipTap outputs (currently). If you enter a mention in the current dialpad message input you'll see it only outputs as text (unstyled). It doesn't actually get styled as a link until it goes through the server and displays as an output chat message.

> * Should I do the same with emoji to keep the same behavior on all plugins?

I'm not quite sure what we should do about emoji... It's a bit different because we have the "emoji-text-wrapper" component, and the emoji outputs using our custom component.

> * Should we update the example to output the HTML instead of plain text so we can copy and paste the output to parse the examples?

Yeah I think so, that's a good idea.

File: packages/dialtone-vue2/components/rich_text_editor/rich_text_editor.vue (hunk 3 above)
Comment (braddialpad) [reply packages/dialtone-vue2/components/rich_text_editor/rich_text_editor.vue]: Hmm.. think I didn't mean to delete this.

File: packages/dialtone-vue2/components/rich_text_editor/rich_text_editor.stories.js (hunk 2 above)
Comment (braddialpad) [reply packages/dialtone-vue2/components/rich_text_editor/rich_text_editor.stories.js]: Don't think it's necessary, removed.

**Comment by braddialpad** [comment]
> Found kind of a bug but I'm not able to reproduce it consistently.

Do you know if it was on Vue 2 or Vue 3 you were getting this problem?

**Comment by juliodialpad** [comment]
> > Found kind of a bug but I'm not able to reproduce it consistently.
> 
> 
> 
> Do you know if it was on Vue 2 or Vue 3 you were getting this problem?

I was testing on Vue 2

**Comment by braddialpad** [comment]
> Sometimes you loose the keyboard navigation ability until you click an option from the dropdown, It happened sometimes when copy/pasting text and then trying to mention a channel/mention but not happening always so not sure if it's a bug or not.

hmm tried a bunch on both vue 2 and vue 3 wasn't able to repro since the recent fixes I made. Let me know if it's still happening for you

File: packages/dialtone-vue3/components/rich_text_editor/rich_text_editor_default.story.vue [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <template>
   <dt-rich-text-editor
-    v-model="$attrs.modelValue"
+    v-model="value"
```
Comment (juliodialpad) [inline packages/dialtone-vue3/components/rich_text_editor/rich_text_editor_default.story.vue:3]: why not using `$attrs.modelValue` directly?

**Review by juliodialpad — APPROVED** [review]
Looking good! just left a comment, please resolve it before merging

File: packages/dialtone-vue3/components/rich_text_editor/rich_text_editor_default.story.vue (hunk 4 above)
Comment (braddialpad) [reply packages/dialtone-vue3/components/rich_text_editor/rich_text_editor_default.story.vue:3]: `$attrs.modelValue` is readonly, and v-model tries to change it. It is common for us to do this with v-model. See `dropdown_default.story.vue` for example.

**Pull request merged**